src/jdk.jlink/share/classes/jdk/tools/jlink/internal/plugins/SystemModulesPlugin.java

```diff
@@ -329,6 +333,17 @@ private String genSystemModulesClass(List<ModuleInfo> moduleInfos,
         SystemModulesClassGenerator generator
             = new SystemModulesClassGenerator(className, moduleInfos, moduleDescriptorsPerMethod);
         byte[] bytes = generator.genClassBytes(cf);
+        // Diagnosis help, can be removed
+        if (Boolean.parseBoolean(System.getProperty("JlinkDumpSystemModuleClass", "false"))) {
```

Comment:
jdk.internal.util is exported to jlink, so you can just use `ClassFileDumper` there.

Reply:
`ClassFileDumper` is not applicable here as it dumps class bytes when it's being defined.  

```suggestion
        if (Boolean.parseBoolean(System.getProperty("jlink.dumpSystemModuleClass", false))) {
```